### PR TITLE
Bump to sdk 2.9.0

### PR DESCRIPTION
**Pull request by eskaur**
Draft to get early feedback.
One of the commits is temporary hack to make it build against the preview installers.

### CODE REVIEW

File: test/test_region_of_interest.py
Comment (eskaur) [inline test/test_region_of_interest.py:56]: If you just wonder what the front-end ROI API looks like, this is the place to look.

File: src/DataModel.cpp
Comment (knzivid) [inline src/DataModel.cpp:38]: I tried and failed getting `pybind11::implicitly_convertible<pybind11::iterable, R>();` to work. If this works, it would cover all three and more (like numpy arrays). Any idea why?

File: src/include/ZividPython/DataModelWrapper.h
Comment (knzivid) [inline src/include/ZividPython/DataModelWrapper.h:257]: I rabbit-holed myself into attributes vs properties in python because I don't use attrs often. TIL while an attribute can be set by the user (`obj.is_array_like = False`), a read-only property can not be. Not asking to change it here, of course.

File: src/DataModel.cpp
Comment (eskaur) [reply src/DataModel.cpp:80]: We are free to change this in the future since this pybind11 generated class is not part of the public zivid-python API at all.
But I could rename it to `RangeDouble` or something if you think that reduces confusion.

File: test/test_region_of_interest.py
Comment (eskaur) [reply test/test_region_of_interest.py:12]: Good question. It works because of the clever fixture system we have here. This test depends on the `"file_camera"` fixture, which depends on the `"application"` fixture. The fixtures keep these things alive with a yield statement:

```
@pytest.fixture(name="application", scope="module")
def application_fixture():
    with zivid.Application() as app:
        yield app
```

File: src/DataModel.cpp
Comment (eskaur) [reply src/DataModel.cpp:38]: I tried that here and it seems to work (it builds and all of the tests pass). Pushed fixup now.

Regarding why you failed to do something similar elsewhere: Maybe you were missing some includes? `#include <pybind11/pybind11.h>` is not always sufficient.

Comment (knzivid) [reply src/DataModel.cpp]: Nice! Not sure what I messed up, probably includes like you said. I don't think I included anything extra.

File: src/DataModel.cpp
Comment (knzivid) [reply src/DataModel.cpp:76]: I don't have a strong preference. As long as we can break the internal names, I am happy with just Range :)

File: continuous-integration/code-generation/datamodel_frontend_generator.py
Comment (knzivid) [inline continuous-integration/code-generation/datamodel_frontend_generator.py:535]: is there a situation where `current_class.is_array_like` is true and `is_leaf` false?

Comment (apartridge) [reply continuous-integration/code-generation/datamodel_frontend_generator.py:535]: @knzivid Eskil is away for 3 weeks, do you think this is a blocker for merging this?

Comment (knzivid) [reply continuous-integration/code-generation/datamodel_frontend_generator.py:535]: No, not if is_array_like in C++ is set to false for non leaf nodes? I am not sure what _array_ here implies, so this suggestion could be wrong.

Comment (knzivid) [reply continuous-integration/code-generation/datamodel_frontend_generator.py:535]: Figured why this exists - this is equivalent to `hasattr(current_class, "is_array_like") and current_class.is_array_like`. The `is_leaf` check assumes the C++ implementation detail that `is_array_like` is only set for leaf nodes. I propose using hasattr or adding a comment about this implementation detail.

Comment (knzivid) [reply continuous-integration/code-generation/datamodel_frontend_generator.py:537]: added comment

**Review by knzivid — DISMISSED** [review]
approving the first two commits. waiting for approval from @apartridge for the third one

**Review by apartridge — APPROVED** [review]
cf4837a8077566f9890f897fa09243dbc4e71c4b looks good.

**Review by apartridge — APPROVED** [review]

**Pull request merged**
